docs/reference/module-api/module-overview.md

```diff
@@ -908,12 +829,146 @@ module.exports = {
 An object of methods that execute queries after any builders have been applied. These functions should use existing query methods or [MongoDB cursor methods](https://docs.mongodb.com/manual/reference/method/js-cursor/) to return documents.
 <!-- TODO: Link to a reference of core builders when available. -->
 
+```javascript
+// modules/product/index.js
+module.exports = {
+  // ...
+  queries(self, query) {
+    return {
+      methods: {
+        // Adds a query method to deliver a random doc that meets the query
+        // criteria.
+        async toRandomObject() {
+          await query.finalize();
+
+          const pipeline = [
+            { $match: query.get('criteria') },
+            { $sample: { size: 1 } }
+          ];
+          const result = await self.apos.doc.db.aggregate(pipeline)
+            .toArray();
+
+          return result[0];
+        }
+      }
+    };
+  }
+};
+```
+
 #### `extendQueries(self)`
 
+Extend the behavior of existing event handlers (set in `queries`) in the `extendQueries` section. This function must return an object as described in `queries`.
+
+Each extended query builder or method should accept the original function as `_super` followed by its original arguments. Extended query builders and methods will be matched with the base class builder or method using the same name. Methods should return data in a similar format to the existing API route.
```

Comment:
It should be mentioend that they need to be nested in `builders` and `methods` sub-sections again.